### ISSUE TITLE
found some incorrect refactors

### ISSUE THREAD

**Issue by ghorn** [ndmitchell/hlint#499]
While refactoring a large codebase, I found a list of bad refactors.
```
- mr13 = clamp' (-1, 1) $ -2.0*(q1*q3 - q0*q2)
+ mr13 = clamp' (-1, 1) 2.0*(q1*q3 - q0*q2)
```
```
- let newNames = map (("test_" ++ ) . show) $ [1,2..numTests] :: [String]
+ let newNames = map (("test_" ++ ) . show) [12..numTests] :: [String]
```
```
-        | newElevation' < -0.5*pi = -0.5*pi
+        | newElevation' < 0.5*pi = -0.5*pi
```

**Comment by ghorn** [comment]
```
- sequenceA $ fmap sum $ sequenceA foo
+ traverse sum sequenceA foo
```
should be
`traverse sum (sequenceA foo)`

**Comment by ndmitchell** [comment]
This is really useful, thanks for taking the time to report them!

Are these errors that HLint gave? Or did you use the automatic refactoring which lead to the errors?

When I tried the file consisting of `test = sequenceA $ fmap sum $ sequenceA foo` it gave me:

```
Sample.hs:3:8: Warning: Use traverse
Found:
  sequenceA $ fmap sum $ sequenceA foo
Why not:
  traverse sum (sequenceA foo)
```

Which is correct.

For the `newNames` refactoring, I think that is correct - `$` in Haskell doesn't have a lower fixity than `::` so the two expressions are equivalent (as far as I can see).

For `mr13` and `newElevation'` running that expression I don't get any hints.  - was there more surrounding code required to trigger the hint?

**Comment by ghorn** [comment]
You're welcome. The newnames problem is changing `[1,2..numTests]` to `[12..numTests]` which I have since realized is a known issue.

I'm using the automatic refactor. I can go back in the history and reproduce the `mr13` error if you can't reproduce it. It's the scariest one IMO

**Comment by ndmitchell** [comment]
Reproducing mr13 would be super helpful. Agree it's scary, and I can't see why it has any hint, so may well be an HLint proper bug. 

The others are all refactor only bugs. Rather than fix one by one, I plan to run the full test suite with refactor, which should shake all of them out. That will take a bit longer, since I'm not really familiar with the refactor part.

**Comment by ghorn** [comment]
ok
```haskell
-- Foo.hs
y = f $ -2.0*x
```
`hlint --refactor --refactor-options="--inplace" Foo.hs`
yields
```haskell
-- Foo.hs
y = f $ 2.0*x
```

**Comment by ghorn** [comment]
```
$ hlint --version
HLint v2.1.5, (C) Neil Mitchell 2006-2018
```

**Comment by ghorn** [comment]
It only happens with refactor, hlint itself gives no hints

**Comment by ndmitchell** [comment]
Weird, never seen a bug where refactor just messes up something entirely separate. So they are all refactor bugs, good for HLint proper, and I'll sweep them up when I add refactor tests. The effort reporting them has been appreciated!

**Comment by ndmitchell** [comment]
Covered by #504
